services/venice-router/src/main/java/com/linkedin/venice/router/MetaDataHandler.java

```diff
@@ -302,7 +299,6 @@ private void hanldeD2ServiceLookup(ChannelHandlerContext ctx, VenicePathParserHe
       responseObject.setName(config.get().getStoreName());
       responseObject.setD2Service(d2Service);
       responseObject.setZkAddress(zkAddress);
-      responseObject.setKafkaZkAddress(kafkaZkAddress);
```

Comment:
Will the client gracefully handle if this is missing from the response?

Reply:
IMO it should get `null` value when it is not set. And it should be ok since we don't seem to be using it.

Let me remove unused kafka zk address from `D2ServiceDiscoveryResponseV2`. Thanks

Reply:
Can you test this locally? No need to add to the PR

Create a class `OldD2ServiceDiscoveryResponseV2` copied from the older `D2ServiceDiscoveryResponseV2`. Use jackson to serialize `D2ServiceDiscoveryResponseV2` object and deserialize to `OldD2ServiceDiscoveryResponseV2`?

Reply:
Sure, let me do it. Thanks

Reply:
Verified. Found no issues when `OldD2ServiceDiscoveryResponseV2` desers response sent with `D2ServiceDiscoveryResponseV2`.